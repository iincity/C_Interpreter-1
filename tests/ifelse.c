#include <stdio.h>

int a = 1;



int main()
{
	int local = 54;
	int test1 = 1, test2 = 2, test3 = 3;

	if (local == 54)
		printf("%d\n", local);
	else {
		local += 1;
	}
	printf("%d\n", local);

	if (local < 60)
		printf("%d\n", test1);
	if (local < 55)
		printf("%d\n", test2);
	else
		printf("%d\n", test3);
		
	if (local > 60)
		printf("%d\n", test1);
	if (local > 55)
		printf("%d\n", test2);
	else
		printf("%d\n", test3);
	
	if (local -= local) {
		int a = 123;
		printf("%d\n", a);
		a + 2;
	}

	if (!local) {
		if (test3 == 3)
			printf("%d\n", test3);
		else
			printf("%d\n", test2);
		if (test1 == test3 - test2) {
			printf("%d\n", test1);
			printf("%d\n", local);
		} else {
			int tmp = test1 - 2;
			printf("%d\n", tmp);
		}
	} else {
		local += test1 * test2 * test3;
		printf("%d\n", local);
	}
	
	local = !local;
	if (!local) {
		if (test3 == 3)
			printf("%d\n", test3);
		else
			printf("%d\n", test2);
		if (test1 == test3 - test2) {
			printf("%d\n", test1);
			printf("%d\n", local);
		} else {
			int tmp = test1 - 2;
			printf("%d\n", tmp);
		}
	} else {
		local += test1 * test2 * test3;
		printf("%d\n", local);
	}


	return 0;
}


