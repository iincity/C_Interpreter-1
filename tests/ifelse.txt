

int a = 1;



int main()
{
	int local = 54;
	int test1 = 1, test2 = 2, test3 = 3;

	if (local == 54)
		print local;
	else {
		local += 1;
	}
	print local;

	if (local < 60)
		print test1;
	if (local < 55)
		print test2;
	else
		print test3;
		
	if (local > 60)
		print test1;
	if (local > 55)
		print test2;
	else
		print test3;
	
	if (local -= local) {
		int a = 123;
		print a;
		a + 2;
	}

	if (!local) {
		if (test3 == 3)
			print test3;
		else
			print test2;
		if (test1 == test3 - test2) {
			print test1;
			print local;
		} else {
			int tmp = test1 - 2;
			print tmp;
		}
	} else {
		local += test1 * test2 * test3;
		print local;
	}
	
	local = !local;
	if (!local) {
		if (test3 == 3)
			print test3;
		else
			print test2;
		if (test1 == test3 - test2) {
			print test1;
			print local;
		} else {
			int tmp = test1 - 2;
			print tmp;
		}
	} else {
		local += test1 * test2 * test3;
		print local;
	}


	return 0;
}


