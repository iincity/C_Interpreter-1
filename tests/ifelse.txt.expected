54
54
1
2
3
3
1
0
7
